oracle/types/market.go

```diff
@@ -1,160 +1,66 @@
+	// provider configurations that match the provider name.
+	for _, market := range marketMap.Markets {
+		for _, cfg := range market.ProviderConfigs {
+			if cfg.Name != name {
+				continue
+			}
+			if _, ok := seenOffChainTickers[cfg.OffChainTicker]; ok {
 				continue
 			}
 
-			tickers[ticker] = provider
-			break
-		}
-	}
-
-	return NewProviderMarketMap(name, tickers)
-}
-
-// NewProviderMarketMap returns a new provider market map.
-func NewProviderMarketMap(name string, tickerConfigs TickerToProviderConfig) (ProviderMarketMap, error) {
-	if len(name) == 0 {
-		return ProviderMarketMap{}, fmt.Errorf("provider name cannot be empty")
-	}
-
-	if len(tickerConfigs) == 0 {
-		return ProviderMarketMap{
-			Name:          name,
-			TickerConfigs: make(map[mmtypes.Ticker]mmtypes.ProviderConfig),
-			OffChainMap:   make(map[string]mmtypes.Ticker),
-		}, nil
-	}
-
-	offChainMap := make(map[string]mmtypes.Ticker)
-	for ticker, config := range tickerConfigs {
-		if err := ticker.ValidateBasic(); err != nil {
-			return ProviderMarketMap{}, fmt.Errorf("invalid ticker %s: %w", ticker, err)
-		}
-
-		if err := config.ValidateBasic(); err != nil {
-			return ProviderMarketMap{}, fmt.Errorf("invalid provider config for %s: %w", ticker, err)
-		}
-
-		if config.Name != name {
-			return ProviderMarketMap{}, fmt.Errorf("expected provider config name %s, got %s", name, config.Name)
-		}
-
-		offChainMap[config.OffChainTicker] = ticker
-	}
-
-	return ProviderMarketMap{
-		Name:          name,
-		TickerConfigs: tickerConfigs,
-		OffChainMap:   offChainMap,
-	}, nil
-}
-
-// GetTickers returns the tickers from the provider market map.
-func (pmm *ProviderMarketMap) GetTickers() []mmtypes.Ticker {
-	tickers := make([]mmtypes.Ticker, 0, len(pmm.TickerConfigs))
-	for ticker := range pmm.TickerConfigs {
-		tickers = append(tickers, ticker)
-	}
-	return tickers
-}
-
-// ValidateBasic performs basic validation on the provider market map.
-func (pmm *ProviderMarketMap) ValidateBasic() error {
-	if len(pmm.Name) == 0 {
-		return fmt.Errorf("provider name cannot be empty")
-	}
-	if len(pmm.OffChainMap) != len(pmm.TickerConfigs) {
-		return fmt.Errorf("off-chain map length mismatch")
-	}
-
-	for ticker, config := range pmm.TickerConfigs {
-		if err := ticker.ValidateBasic(); err != nil {
-			return fmt.Errorf("invalid ticker %s: %w", ticker, err)
-		}
-
-		if err := config.ValidateBasic(); err != nil {
-			return fmt.Errorf("invalid provider config for %s: %w", ticker, err)
-		}
-
-		t, ok := pmm.OffChainMap[config.OffChainTicker]
-		if !ok {
-			return fmt.Errorf("off-chain ticker %s not found in off-chain map", config.OffChainTicker)
-		}
-
-		if t != ticker {
-			return fmt.Errorf("off-chain ticker %s does not match on-chain ticker %s", config.OffChainTicker, ticker)
+			providerTicker := NewProviderTicker(
+				cfg.Name,
+				cfg.OffChainTicker,
+				cfg.Metadata_JSON,
+				DefaultTickerDecimals,
```

Comment:
Why are we using a default TickerDecimals here? Should this even be relevant to providers if we're moving to `big.Float` arithmetic in providers?

Reply:
hmm, yea i guess scaling up the values without a notion of the necessary precision before hitting aggregation probably just adds overhead.